.github/workflows/process_new_pull_request.yml

```
@@ -0,0 +1,156 @@
+            // INPUT SANITIZATION
+            // ======================
+            const safeGet = (obj, path, fallback = null) => {
+              return path.split('.').reduce((acc, key) => 
+                (acc && acc[key] !== undefined) ? acc[key] : fallback, obj
+              );
+            };
+
+            // Core PR metadata with validation
+            const prContext = context.payload.pull_request || {};
+            const repositoryContext = context.repo;
+            
+            // Validate repository context
+            const repoOwner = (repositoryContext.owner || '').replace(/[^a-zA-Z0-9-]/g, '');
+            const repoName = (repositoryContext.repo || '').replace(/[^a-zA-Z0-9-]/g, '');
+            if (!repoOwner || !repoName) {
+              throw new Error('Invalid repository context');
+            }
+
+            // Validate PR number
+            const prNumber = Number.isInteger(prContext.number) ? prContext.number : 0;
+            if (prNumber <= 0) {
+              throw new Error(`Invalid PR number: ${prNumber}`);
+            }
+
+            // Sanitize PR author
+            const prAuthor = safeGet(prContext, 'user.login', '') || '';
+            const sanitizedAuthor = prAuthor.replace(/[^a-zA-Z0-9_-]/g, '');
+
+            // ======================
+            // CONTRIBUTOR CHECK
+            // ======================
+            const isNewContributor = await (async () => {
+              try {
+                const encodedAuthor = encodeURIComponent(`user:${sanitizedAuthor}`);
+                const searchResults = await github.rest.search.issuesAndPullRequests({
+                  q: `repo:${repoOwner}/${repoName} type:pr is:merged ${encodedAuthor}`
+                });
+                return safeGet(searchResults, 'data.total_count', 0) === 0;
+              } catch (error) {
+                console.error('Contributor check failed:', error.message);
+                return false;
```

as I think I've mentioned on another of your PRs, unless an error is expected under normal circumstances, we should fail noisily rather than suppressing the error with a try-catch

---

Done